compiler/rustc_ast_pretty/src/pprust/state/expr.rs

```diff
@@ -422,7 +422,8 @@ impl<'a> State<'a> {
                 self.print_type(ty);
             }
             ast::ExprKind::Type(expr, ty) => {
-                self.word("type_ascribe!(");
+                self.word("builtin # type_ascribe");
```

Comment:
Hmm... I think I'm kind of torn here. I feel like it's still useful to print it like `type_ascribe!(`, though it's kind of lying to the user. We do the macro-ish syntax for deref pats, since users won't be writing `builtin # deref` themselves.

Reply:
Oh, lol, I see. It's due to #124619.

In that case, I wonder if we need to fix deref patterns too here...

Reply:
I cc'ed #124619 to give some extra context but it's not really related. #124619 describes a `rustfmt` bug which I plan on fixing out of tree / in the `rustfmt` repo.

Reply:
oh, in that case, I feel like we might want to keep this as `type_ascribe!()` lol -- but whatever, if you don't agree, i don't particularly care

Reply:
I'm definitely not a fan of `builtin #`'s visuals, it looks grotesque and verbose. However, printing a fake macro call I like even less.

Reply:
I think it better to print the builtin

Reply:
because that doesn't need to be imported